tests/PHPStan/Analyser/nsrt/get-defined-vars.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace GetDefinedVars;
+
+use function PHPStan\Testing\assertType;
+
+function doFoo(int $param) {
+	$local = "foo";
+	assertType('array{param: int, local: \'foo\'}', get_defined_vars());
+	assertType('array{\'param\', \'local\'}', array_keys(get_defined_vars()));
```

Comment:
I think a test with conditionally defined variables would be helpful

Reply:
Good idea. I added that.